src/shared/parse-helpers.c

```diff
@@ -40,8 +40,19 @@ int path_simplify_and_warn(
                                           lvalue, fatal ? "" : ", ignoring", path);
         }
 
+        if (keep_trailing_slash)
+                keep_trailing_slash = endswith(path, "/");
+
         path_simplify(path);
```

Comment:
Maybe add a new variant to path_simplify? There might be more cases like this in the future

Reply:
My urge to do a quick hack got the better of me, did it properly now

Reply:
Quick hacks are only permissible on Fridays, but today is Thursday